sbncode/TPCReco/VertexStub/TwoPlaneStubMerge_tool.cc

```diff
@@ -0,0 +1,220 @@
+/**
+ *
+ */
+
+// Framework Includes
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "art/Persistency/Common/PtrMaker.h"
+#include "art/Utilities/ToolMacros.h"
+#include "cetlib/cpu_timer.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+// local includes
+#include "sbncode/TPCReco/VertexStub/IStubMerge.h"
+#include "sbncode/TPCReco/VertexStub/StubMergeAlgorithms.h"
+
+// LArSoft includes
+#include "larcorealg/Geometry/BoxBoundedGeo.h"
+#include "larcore/Geometry/Geometry.h"
+#include "larcorealg/Geometry/GeometryCore.h"
+#include "larcore/CoreUtils/ServiceUtil.h"
+
+// std includes
+#include <string>
+#include <iostream>
+#include <memory>
+
+namespace sbn{
+/**
+ *  @brief  TwoPlaneStubMerge class definiton
+ */
+class TwoPlaneStubMerge : public sbn::IStubMerge {
+public:
+    /**
+     *  @brief  Constructor
+     */
+    TwoPlaneStubMerge(fhicl::ParameterSet const &pset);
+
+    /**
+     *  @brief  Destructor
+     */
+    ~TwoPlaneStubMerge();
+
+    std::vector<sbn::StubInfo> Merge(const std::vector<sbn::StubInfo> &stubs,
+       const geo::GeometryCore *geo, 
+       const spacecharge::SpaceCharge *sce, 
+       const detinfo::DetectorClocksData &dclock,
+       const detinfo::DetectorPropertiesData &dprop) override;
+
+    sbn::StubInfo MergeStubs(const sbn::StubInfo &A, const sbn::StubInfo &B,
+      const geo::GeometryCore *geo,
+      const spacecharge::SpaceCharge *sce,
+      const detinfo::DetectorPropertiesData &dprop);
+
+    bool SortStubs(const sbn::StubInfo &A, const sbn::StubInfo &B);
+
+private:
+  sbn::PlaneTransform fPlaneTransform;
+  double fMaxMergeTOff;
+  double fMaxMergeQOff;
+  bool fRemoveDuplicateMerges;
+  bool fSaveOldStubs;
+};
+
+TwoPlaneStubMerge::TwoPlaneStubMerge(fhicl::ParameterSet const &pset):
+  fPlaneTransform(pset.get<fhicl::ParameterSet>("PlaneTransform")),
+  fMaxMergeTOff(pset.get<double>("MaxMergeTOff")),
+  fMaxMergeQOff(pset.get<double>("MaxMergeQOff")),
+  fRemoveDuplicateMerges(pset.get<bool>("RemoveDuplicateMerges")),
+  fSaveOldStubs(pset.get<bool>("SaveOldStubs"))
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+TwoPlaneStubMerge::~TwoPlaneStubMerge()
+{
+}
```

Comment:
I think it's fine as a placeholder -- is this any different than an implicit destructor?

Reply:
... maybe.
First, you don't declare it `noexcept`, which the default destructor would be. Compiler might be smart enough to figure out though.
Then, there are a lot of rules of C++ about which constructors and assignment operators are implicitly declared when some others are present (e.g. if a user-defined move constructor is present, a copy assignment operator is not generated; and vice versa; if a user-defined destructor is present, the move constructor is not implicitly declared). I am not sure what is the discretion of the compiler here; in general, I prefer to just avoid the mess completely and let the compiler the most free.

Additionally, if you add _content_ in the destructor and I am reviewing the changes, I will then try to persuade you to rewrite the class so that the explicit destruction is not needed. That is possible most of the times in C++.

Reply:
Fixed.